src/loaders/zarrLoader.js

```diff
@@ -116,76 +112,64 @@ export default class ZarrLoader {
   getRasterSize({ z }) {
     const source = this._getSource(z);
-    const height = source.shape[this._yIndex];
-    const width = source.shape[this._xIndex];
+    const [height, width] = ['y', 'x'].map(
+      k => source.shape[this._dimIndices.get(k)]
+    );
     return { height, width };
   }
 
-  /**
-   * Converts Array of loader selection objects into zarr-specific selection
-   *
-   * Ex.
-   *  const loaderSelectionObj = [
-   *     { time: 1, channel: 'a' },
-   *     { time: 1, channel: 'b' }
-   *  ];
-   *  const serialized = loader.serializeSelection(loaderSelectionObj);
-   *  console.log(serialized);
-   *  // [[1, 0, 0, 0], [1, 1, 0, 0]]
-   *
-   * @param {Array || Object} loaderSelectionObjs Human-interpretable array of desired selection objects
-   * @returns {Array} number[][], zarr-specific selections to be passed to to viv as loaderSelections
-   */
-  serializeSelection(loaderSelectionObjs) {
-    // Wrap selection in array if only one is provided
-    const selectionObjs = Array.isArray(loaderSelectionObjs)
-      ? loaderSelectionObjs
-      : [loaderSelectionObjs];
-
-    const serialized = selectionObjs.map(obj => this._serialize(obj));
-    return serialized;
-  }
-
   _getSource(z) {
     return typeof z === 'number' && this.isPyramid ? this._data[z] : this._data;
   }
 
-  _serialize(selectionObj) {
-    const serializedSelection = Array(this.dimensions.length).fill(0);
-    const dimFields = this.dimensions.map(d => d.field);
-    Object.entries(selectionObj).forEach(([key, val]) => {
-      // Get index of named dimension in zarr array
-      const dimIndex = dimFields.indexOf(key);
-      if (dimIndex === undefined) {
-        throw Error(
-          `Dimension '${key}' does not exist on array with dimensions :
-          ${dimFields}`
-        );
-      }
-      // Get position of index along dimension axis
-      let valueIndex;
-      const { field, type, values } = this.dimensions[dimIndex];
-      if (typeof val === 'number') {
-        // Assign index directly, regardless of dimension type
-        valueIndex = val;
-      } else if (type === 'ordinal' || type === 'nominal') {
-        // Lookup index if categorical dimension type.
-        // This is slower if dimension is large; setting directly is preferred.
-        valueIndex = values.indexOf(val);
-      } else {
-        // Cannot use string value for dimension that is 'quantitative' or 'temporal', must set directly.
+  /**
+   * Returns valid zarr.js selection for ZarrArray.getRaw or ZarrArray.getRawChunk
+   * @param {Object} selection valid dimension selection
+   * @returns {Array} Array of indicies
+   *
+   * Valid dimension selections include:
+   *   - Direct zarr.js selection: [1, 0, 0, 0]
+   *   - Named selection object: { channel: 0, time: 2 } or { channel: "DAPI", time: 2 }
+   */
+  _serializeSelection(selection) {
+    // Just return copy if array-like zarr.js selection
+    if (Array.isArray(selection)) return [...selection];
```

Comment:
Why do you want to support this?

Reply:
The loader has a default selection on it, so that if no selection is specified something still gets returned. I'm not sure if this should be case, and would be willing to remove...

Reply:
So `loaderSelection` should just be one of the objects named selection objects.